packages/discord.js/src/structures/ClientApplication.js

```diff
@@ -68,6 +68,52 @@ class ClientApplication extends Application {
       this.installParams ??= null;
     }
 
+    /**
+     * OAuth2 installation paremeters.
+     * @typedef {Object} IntegrationTypesConfigurationParameters
+     * @property {?OAuth2Scopes[]} scopes Scopes that will be set upon adding this application
+     * @property {?Readonly<PermissionsBitField>} permissions Permissions that will be requested for the integrated role
+     */
+
+    /**
+     * The application's supported installation context data.
+     * @typedef {Object} IntegrationTypesConfigurationContext
+     * @property {?IntegrationTypesConfigurationParameters} oauth2InstallParams
+     * Scopes and permissions regarding the installation context
+     */
+
+    /**
+     * The application's supported installation context data.
+     * @typedef {Object} IntegrationTypesConfiguration
+     * @property {?IntegrationTypesConfigurationContext} 0 Scopes and permissions
+     * regarding the guild-installation context
+     * @property {?IntegrationTypesConfigurationContext} 1 Scopes and permissions
+     * regarding the user-installation context
+     */
+
+    if ('integration_types_config' in data) {
+      /**
+       * Default scopes and permissions for each supported installation context.
+       * The keys are stringified variants of {@link ApplicationIntegrationType}.
+       * @type {IntegrationTypesConfiguration}
+       */
+      this.integrationTypesConfig = Object.fromEntries(
+        Object.entries(data.integration_types_config).map(([key, config]) => {
+          const context = {
+            oauth2InstallParams: {
+              scopes: config.oauth2_install_params?.scopes ?? null,
+              permissions: config.oauth2_install_params
+                ? new PermissionsBitField(config.oauth2_install_params.permissions).freeze()
+                : null,
+            },
```

Comment:
Can both `scopes` and `permissions` be null at the same time?
If so, in that case, let's make the context `null`, and make `oauth2InstallParams` not nullable? Alternatively, make `oauth2InstallParams` `null`